src/Workspaces/Core/Portable/Workspace/Host/HostProjectServices.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    // TODO(cyrusn): Make public.  Tracked through https://github.com/dotnet/roslyn/issues/62914
+    internal sealed class HostProjectServices
+    {
+        private readonly HostLanguageServices _services;
+
+        public HostSolutionServices SolutionServices => _services.WorkspaceServices.SolutionServices;
+
+        // This ensures a single instance of this type associated with each HostLanguageServices.
+        [Obsolete("Do not call directly.  Use HostLanguageServices.ProjectServices to acquire an instance")]
+        internal HostProjectServices(HostLanguageServices services)
```

Comment:
yes.  but then i have to override `==` and validate all cases whre code checks identity are correct.

Reply:
I don't think you need to override `==`. Struct does not have to have value equality.

Reply:
currently we have code that checks if the Services instances are the same instance.  I wanted to avoid having to go find and audit all of those (esp. as we have no facilty for searching for use of `reference equality == on these types`. :)